Automattic/wp-calypso
# Site Settings: Refactor and improve StartOver component

This PR refactors the `StartOver` component in Settings, particularly suggesting the following:

* Refactoring the component to a stateless functional component.
* Connecting the component to Redux and using `getSelectedSiteSlug()` instead of  `sitesList.getSelectedSite()`.
* Removing the `sitesList` prop dependency from the component.
* ES6ifying the component.
* Moving the SVG icon to a separate file. This also fixes a warning that the `fill-opacity` SVG prop was causing.
* Removing various ESLint warnings.
* Updating the `HeaderCake` to use `backHref` instead of `page()`, removing the `page` dependency.

To test:
* Checkout this branch
* Go to `/settings/start-over/$site`, where `$site` is one of your WordPress.com sites.
* Verify the Back button takes you back to `/settings/general/$site`.
* Verify there are no visual differences or functional changes (besides the fill-opacity of the logo, which was previously not working properly).
* Verify going to `/settings/start-over/$site` for a Jetpack site still takes you to `/settings/general/$site`.
* Verify there is no warning when you visit `/settings/start-over/$site` caused by the `StartOver` component.
